mktrayicon.c

```diff
@@ -19,12 +19,39 @@
  */
 char* strncpy_esc(char *dest, const char *src, size_t n)
 {
-	size_t i;
+	size_t i = 0;
 	size_t index = 0;
-	for (i = 0; i < n && src[i] != '\0'; i++){
-		if (src[i] != '\\'){
+	while (i < n && src[i] != '\0'){
+		if (src[i] == '\''){ 
+			/* 
+			 * this a real single quote; can't have been escaped
+			 * so we move accordingly, copying everything
+			 */
+			i++;
+			while (src[i] != '\''){
```

Comment:
I believe this needs a `&& src[i] != '\0'` clause

Reply:
At first glance it does. I thought about it while writing it. But to call this function we have made sure that number of quotes of `src` is even. So really, if we are in there it means that we will **surely** find the other quote to end the sequence. For safety, and to be consistent, we can add it though.